src/WebJobs.Extensions.DurableTask/EtwEventSource.cs

```diff
@@ -26,14 +26,20 @@ private EtwEventSource()
             string AppName,
             string SlotName,
             string FunctionName,
-            string Version,
             string InstanceId,
             string Reason,
             string FunctionType,
             string ExtensionVersion,
             bool IsReplay)
         {
-            this.WriteEvent(201, TaskHub, AppName, SlotName, FunctionName, Version ?? "", InstanceId, Reason, FunctionType, ExtensionVersion, IsReplay);
+            if (!string.IsNullOrEmpty(AppName) && !string.IsNullOrEmpty(SlotName))
+            {
+                this.WriteEvent(201, TaskHub, AppName, SlotName, FunctionName, InstanceId, Reason, FunctionType, ExtensionVersion, IsReplay);
+            }
+            else
+            {
+                this.WriteEvent(201, TaskHub, FunctionName, InstanceId, Reason, FunctionType, ExtensionVersion, IsReplay);
+            }
```

Comment:
This change will actually break our ETW telemetry because the `WriteEvent` parameters must match the method signature exactly. Instead, let's leave it as it was, but simply remove the `Version` parameter. To be safe, we should also increment the version of this ETW event by updating the `[Event]` attribute as follows:

```csharp
[Event(201, Level = EventLevel.Informational, Version = 2)]
```
Please apply similar changes to the other methods in this class.